p2p/switch.go

```diff
@@ -369,21 +369,44 @@ func (sw *Switch) IsDialing(addr *NetAddress) bool {
 
 // Broadcast runs a go routine for each attempted send, which will block
 // trying to send for defaultSendTimeoutSeconds. Returns a channel
-// which receives success values for each attempted send (false if times out).
+// which receives broadcast result for each attempted send (success=false if times out).
 // NOTE: Broadcast uses goroutines, so order of broadcast may not be preserved.
 // TODO: Something more intelligent.
-func (sw *Switch) Broadcast(chID byte, msg interface{}) chan bool {
-	successChan := make(chan bool, len(sw.peers.List()))
+
+type BroadcastResult struct {
+	PeerKey string
+	Success bool
+}
+
+func (sw *Switch) Broadcast(chID byte, msg interface{}) chan BroadcastResult {
+	successChan := make(chan BroadcastResult, len(sw.peers.List()))
 	sw.Logger.Debug("Broadcast", "channel", chID, "msg", msg)
 	for _, peer := range sw.peers.List() {
 		go func(peer Peer) {
```

Comment:
#805 we are planning to get rid of goroutines

Reply:
Right. The idea was maybe we'd have some long running "worker" routines that handle our broadcasts, instead of spawning a new routine for each peer every time.

Reply:
Oh, that said, the approach in TryBroadcast is kind of interesting. Perhaps we should consider that. Probably best if we move it into a separate PR so we can discuss it independent of the pong work.